src/formio/components/Partners/Partners.stories.js

```diff
@@ -0,0 +1,137 @@
+import {expect, fn, screen, userEvent, within} from '@storybook/test';
+
+import {ConfigDecorator} from 'story-utils/decorators';
+
+import {SingleFormioComponent} from '../story-util';
+
+export default {
+  title: 'Form.io components / Custom / Partners',
+  decorators: [ConfigDecorator],
+  render: SingleFormioComponent,
+  args: {
+    type: 'partners',
+    key: 'partners',
+    label: 'Partners',
+  },
+};
+
+export const WithDataRetrieved = {
+  args: {
+    submissionData: {
+      partners: [
+        {
+          bsn: '123456789',
+          initials: 'J',
+          affixes: 'K',
+          lastName: 'Boei',
+          dateOfBirth: '1-1-2000',
+          __addedManually: false,
+        },
+        {
+          bsn: '165456987',
+          initials: '',
+          affixes: '',
+          lastName: 'Tijest',
+          dateOfBirth: '1-1-1990',
+          __addedManually: false,
+        },
+      ],
+    },
+  },
+};
+
+export const NoDataRetrieved = {};
+
+export const ManuallyAddedPartner = {
+  args: {
+    submissionData: {
+      partners: [
+        {
+          bsn: '123456789',
+          initials: 'J',
+          affixes: 'K',
+          lastName: 'Boei',
+          dateOfBirth: '1-1-2000',
+          __addedManually: true,
+        },
+      ],
+    },
+  },
+};
+
+export const EditManuallyAddedPartner = {
+  args: {
+    onSave: fn(),
+    submissionData: {
+      partners: [
+        {
+          bsn: '123456782',
+          initials: 'J',
+          affixes: 'K',
+          lastName: 'Boei',
+          dateOfBirth: '2000-1-1',
+          __addedManually: true,
+        },
+      ],
+    },
+  },
+  play: async ({canvasElement, step}) => {
+    const canvas = within(canvasElement);
+    const user = userEvent.setup();
+
+    await step('Check partners details', async () => {
+      const bsnLabel = await canvas.findByText(
+        (content, element) => element?.tagName.toLowerCase() === 'dt' && content.trim() === 'BSN'
+      );
+      const bsnValue = bsnLabel.nextElementSibling;
+      expect(bsnValue?.textContent).toBe('123456782');
```

Comment:
I'm not too happy about this kind of test code, but I don't have the energy now to look for better alternatives. Let's not repeat stuff like this in the future, yeah? :stuck_out_tongue: